url0d38a10ee7_diegopliebana_GEBT_tgz-pJ8-ch_idsia_scenarios_test_EvolveWithChangingSeedsJ8/src/ch/idsia/mario/engine/mapedit/LevelEditor.java

```diff
@@ -0,0 +1,151 @@
+package ch.idsia.mario.engine.mapedit;
+
+import ch.idsia.mario.engine.LevelScene;
+import ch.idsia.mario.engine.level.Level;
+
+import javax.swing.*;
+import javax.swing.border.EtchedBorder;
+import javax.swing.border.TitledBorder;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.FileOutputStream;
+
+
+public class LevelEditor extends JFrame implements ActionListener
+{
+    private static final long serialVersionUID = 7461321112832160393L;
+
+    private JButton loadButton;
+    private JButton saveButton;
+    private JTextField nameField;
+    private LevelEditView levelEditView;
+    private TilePicker tilePicker;
+    
+    private JCheckBox[] bitmapCheckboxes = new JCheckBox[8];
+
+    public LevelEditor()
+    {
+        super("Map Edit");
+        
+        try
+        {
+            System.out.println("System.getProperty(\"user.dir()\") = " + System.getProperty("user.dir"));
+            Level.loadBehaviors(new DataInputStream(LevelScene.class.getResourceAsStream("resources/tiles.dat")));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *DataInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when Level.loadBehaviors throws an exception. Ensure the object returned by *DataInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).